examples/thermostat/thermostat-common/include/atomic-write-manager.h

```diff
@@ -0,0 +1,186 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/thermostat-server/atomic-write.h>
+#include <app/clusters/thermostat-server/thermostat-delegate.h>
+
+#include "thermostat-delegate-impl.h"
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Thermostat {
+
+enum AtomicWriteState
+{
+    kAtomicWriteState_Closed = 0,
+    kAtomicWriteState_Open,
```

Comment:
do we need to duplicate the AtomicWriteState here? how about kClosed kOpen. so we can use AtomicWriteState::kOpen or AtomicWriteState::kClosed.